app/models/user.rb

```diff
@@ -21,7 +21,8 @@ class User < ActiveRecord::Base
   validates :password,              presence:     { if: :requires_password? }
 
   validates :email,                 presence:     { if: :email_required? },
-                                    uniqueness:   { if: :email_required_or_present? }
+                                    uniqueness:   { if: :email_required_or_present? },
+                                    format:       { if: :email_required_or_present?, with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i}
 
```

Comment:
@marcellosachs Don't use this regex to validate email addresses. There is hopefully a gem or built-in rails feature that does that. The actual regex used in the RFC is surprisingly complex:

http://www.ex-parrot.com/pdw/Mail-RFC822-Address.html